### PR TITLE
fix deprecation warnings in tests

### PR DESCRIPTION
**Pull request by tjhei**
we do ignore them, but this will break at some point if we don't fix
them.

### CODE REVIEW

File: tests/signals_connect.cc [hunk 1]
```diff
@@ -8,7 +8,7 @@ using namespace aspect;
 
 template <int dim>
 void post_constraints_creation (const SimulatorAccess<dim> &,
-                                ConstraintMatrix &)
+                                aspect::ConstraintMatrix &)
```
Comment (bangerth) [inline tests/signals_connect.cc:11]: Can you revert this change and the corresponding changes in the other two files? I'm removing all of these uses of the class in #3738.

**Comment by tjhei** [comment]
done
